src/webring.rs

```diff
@@ -74,25 +79,53 @@ pub struct Webring {
     members_file_path: PathBuf,
     static_dir_path: PathBuf,
     file_watcher: OnceLock<RecommendedWatcher>,
-    base_address: Uri,
+    base_address: Intern<Uri>,
+    base_authority: Intern<str>,
+    stats: Arc<Stats>,
+}
+
+// Change this if Henry figures out how to make a sensible `Default` implementation for `Intern<*something unsized*>` in sarlacc
+impl Default for Webring {
+    fn default() -> Self {
+        Webring {
+            inner: RwLock::default(),
+            homepage: tokio::sync::RwLock::default(),
+            members_file_path: PathBuf::default(),
+            static_dir_path: PathBuf::default(),
+            file_watcher: OnceLock::new(),
+            base_address: Intern::default(),
+            base_authority: Intern::from_ref("ring.purduehacker.com"),
```

Comment:
Typo: should be `ring.purduehackers.com`.

Also, since we're hard-coding the domain now, can we move the `Default` impl into `mod tests` as that's the only place it gets used? For the actual application we call `Webring::new()` and provide the base address.